### PR TITLE
Update the cross-validation to be able to use user-inputted groups

### PR DESCRIPTION
**Pull request by mwjjeong**
- Linear and logistic regressions were updated to be able to use user-inputted labels when splitting training sets for cross validation.
- The *groups* argument is equivalent to the *foldid* argument of *cvglmnet* in *R*.
- Use *joblib* and *six* modules instead of *sklearn.external*.
- Check the validity of the shape of weights in *logitnet.py*.

### CODE REVIEW

File: glmnet/linear.py [hunk 1]
```diff
@@ -180,9 +178,12 @@ def fit(self, X, y, sample_weight=None, relative_penalties=None):
         X : array, shape (n_samples, n_features)
             Input features
 
-        Y : array, shape (n_samples,)
+        y : array, shape (n_samples,)
             Target values
 
+        groups: array, shape (n_samples,)
+            Group labels for the samples used while splitting the dataset into train/test set.
```
Comment (mheilman) [inline glmnet/linear.py:185]: I think the docstring here should specify what happens when `None` is provided (e.g., "If None, then data will be split randomly for K-fold cross-validation.").

Also, I think this description should mention that the groups will be passed to sklearn.model_selection.GroupKFold. Without a link to that, it seems like somebody would need to look through the code to figure out how this `groups` variable is going to work

**Comment by mheilman** [comment]
thanks for the PR!

**Comment by mheilman** [comment]
Just a note: this is related to #61 since joblib was updated here. A new release is still needed to get these changes out.
